File: backend/src/main/java/org/studysystem/backend/service/GradeService.java
package org.studysystem.backend.service;

import org.studysystem.backend.dto.request.GradeRequest;
import org.studysystem.backend.dto.response.GradeResponse;

import java.util.List;

public interface GradeService {
    void CreateGrade(Long userId, Long courseId, GradeRequest gradeRequest);

    void updateGrade(Long gradeId, GradeRequest gradeRequest);

    List<GradeResponse> getGradeByUserId(Long userId);

    List<GradeResponse> getGradeByCourseId(Long courseId);
}

File: frontend/src/services/GradeService.js
import api from "../AxiosConfig";

const REST_API_BASE_URL = "/grade";

export const createGrade = (userId, courseId, grade) => api.post(REST_API_BASE_URL + "/user/" + userId + "/course/" + courseId, grade);

export const updateGrade = (gradeId, grade) => api.put(`${REST_API_BASE_URL}/${gradeId}`, grade);

export const getGradeByUserId = (userId) => api.get(`${REST_API_BASE_URL}/user/${userId}`);

export const getGradeByCourseId = (courseId) => api.get(`${REST_API_BASE_URL}/course/${courseId}`);